src/icons/test.ts

```diff
@@ -0,0 +1,55 @@
+import path from 'path';
+import fs from 'fs';
+
+/**
+ * Transform to help us compare name equality between icon component files and their
+ * svg counterparts.
+ *
+ * @example
+ * transformFileName("some-icon.svg") === "someicon"
+ * transformFileName("someIcon.ts") === "someicon"
+ */
+const transformFileName = (fileName: string) =>
+  fileName.toLowerCase().replace(/(-|.ts$|.svg$|)/g, '');
+
+const SVG_DIRECTORY_NAME = 'svgs';
+
+const testIconDirectory = (directory: string) => {
+  it(`should have the same number of ${directory} as svg paths, with complementary naming`, () => {
+    const iconFilenames = fs
+      .readdirSync(path.resolve(__dirname, directory))
+      .filter((file) => {
+        // Ignore svg directory
+        if (file === SVG_DIRECTORY_NAME) return false;
+        // Ignore exports file
+        if (file === 'index.ts') return false;
+
+        return true;
+      })
+      .map(transformFileName);
+
+    const svgFilenames = fs
+      .readdirSync(
+        path.resolve(__dirname, `${directory}/${SVG_DIRECTORY_NAME}`),
+      )
+      .map(transformFileName);
+
+    expect(iconFilenames.sort()).toEqual(svgFilenames.sort());
+  });
+};
+
+const ICONS_MAP = {
+  emojis: 'emojis',
+  glyphs: 'glyphs',
+  icons: 'icons',
+  logos: 'logos',
+  navIcons: 'navIcons',
+};
+
+describe('icons', () => {
```

Comment:
Since there is more manual effort involved in adding/maintaining existing icons, this test at least ensures that typos do not inadvertently break expected functionality.